commands/azureCommands/pull-from-azure.ts

```diff
@@ -0,0 +1,38 @@
+import vscode = require('vscode');
+import { reporter } from '../../telemetry/telemetry';
+const teleCmdId: string = 'vscode-docker.image.pullFromAzure';
+import { AzureImageNode } from '../../explorer/models/AzureRegistryNodes';
+import * as acrTools from '../../utils/Azure/acrTools';
+
+/* Pulls an image from Azure. The context is the image node the user has right clicked on */
+export async function pullFromAzure(context?: AzureImageNode): Promise<any> {
+
+    // Step 1: Using loginCredentials() function to get the username and password. This takes care of all users, even if they don't have the Azure CLI
+    let credentials;
+    try {
+        credentials = await acrTools.loginCredentialsRefreshToken(context.subscription, context.registry, context);
+    } catch (error) {
+        console.log(error);
+    }
+    let username = credentials.username;
+    let password = credentials.password;
+    let registry = context.registry.loginServer;
+
+    const terminal = vscode.window.createTerminal("Docker");
+    terminal.show();
+
+    // Step 2: docker login command
+    terminal.sendText(`docker login ${registry} -u ${username} -p ${password}`);
```

Comment:
Will terminal.sendText wait for the command finish? I think the following `docker pull` command will rely on it.

Reply:
Yes, docker pull relies on that. Will putting await before each command take care of that?

Reply:
I don't see other codes invoking docker command in terminal use await. But if sendText returns IPromise, I assume it's no harm to call await.

---
In reply to: [210086973](https://github.com/AzureCR/vscode-docker/pull/53#discussion_r210086973) [](ancestors = 210086973)